src/redis-cli.c

```diff
@@ -1216,20 +1223,21 @@ static sds cliFormatReplyJson(sds out, redisReply *r) {
         out = sdscat(out,"{");
         for (i = 0; i < r->elements; i += 2) {
             redisReply *key = r->element[i];
-            if (key->type == REDIS_REPLY_STATUS ||
+            if (key->type == REDIS_REPLY_ERROR ||
+                key->type == REDIS_REPLY_STATUS ||
                 key->type == REDIS_REPLY_STRING ||
                 key->type == REDIS_REPLY_VERB) {
-                out = cliFormatReplyJson(out, key);
+                out = cliFormatReplyJson(out, key, JSON_DISABLE_BINARY_ENCODING);
             } else {
                 /* According to JSON spec, JSON map keys must be strings, */
                 /* and in RESP3, they can be other types. */
-                sds tmp = cliFormatReplyJson(sdsempty(), key);
-                out = sdscatrepr(out,tmp,sdslen(tmp));
+                sds tmp = cliFormatReplyJson(sdsempty(), key, JSON_DISABLE_BINARY_ENCODING);
+                out = escapeJsonString(out,tmp,sdslen(tmp));
```

Comment:
Isn't the output of the previous line already escaped? in which case all we need is add quotes? 
I'm not sure..

Reply:
The first one is to convert non-string type to string since JSON key should be string, the second one is to escape the converted string.
I'll comment it in the source code as well.